packages/salesforcedx-vscode-soql/src/commonUtils.ts

```diff
@@ -6,23 +6,39 @@
  */
 
 import * as path from 'path';
-import { TextDocument, workspace, WorkspaceFolder } from 'vscode';
+import * as vscode from 'vscode';
+import { channelService } from './channel';
+import { telemetryService } from './telemetry';
 
-export function getDocumentName(document: TextDocument): string {
+export function getDocumentName(document: vscode.TextDocument): string {
   const documentPath = document.uri.fsPath;
   return path.basename(documentPath) || '';
 }
 
-function hasRootWorkspace(ws: typeof workspace = workspace) {
+function hasRootWorkspace(ws: typeof vscode.workspace = vscode.workspace) {
   return ws && ws.workspaceFolders && ws.workspaceFolders.length > 0;
 }
 
-function getRootWorkspace(): WorkspaceFolder {
+function getRootWorkspace(): vscode.WorkspaceFolder {
   return hasRootWorkspace()
-    ? workspace.workspaceFolders![0]
-    : ({} as WorkspaceFolder);
+    ? (vscode.workspace.workspaceFolders as vscode.WorkspaceFolder[])[0]
+    : ({} as vscode.WorkspaceFolder);
 }
 
 export function getRootWorkspacePath(): string {
   return getRootWorkspace().uri ? getRootWorkspace().uri.fsPath : '';
 }
+
+export async function trackErrorWithTelemetry(
+  problemId: string,
+  error: string
+): Promise<void> {
+  try {
+    await telemetryService.sendException(
+      `soql_error_${problemId.toLocaleLowerCase()}`,
+      error
+    );
```

Comment:
The telemetry "problem_id" field showing in app insights looks like this.

<img width="397" alt="app-insights-exceptions" src="https://user-images.githubusercontent.com/599418/101824898-1ab8a980-3aea-11eb-8d68-82d83cad85f6.png">

Reply:
nice